### PR TITLE
Add error handling when the call to 'install-url' endpoint fails

### PR DESCRIPTION
**Pull request by bor0**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes https://github.com/Automattic/woocommerce.com/issues/20223.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Apply [test.patch](https://github.com/user-attachments/files/16992941/test.patch) to mimic the error easily
2. Run `pnpm --filter='@woocommerce/plugin-woocommerce' watch:build` and wait for it to build
3. Go to http://localhost:8080/wp-admin/admin.php?page=wc-admin&tab=my-subscriptions&path=%2Fextensions
4. Connect your store if it isn't.
5. Press `Install` on any of the plugins in the list

Screenshot 2024-09-13 at 13 12 20

6. You should get a proper error and the form should remain responsive and continue to work

Screenshot 2024-09-13 at 13 12 35

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/actions/install.tsx [hunk 1]
```diff
@@ -51,6 +51,44 @@ export default function Install( props: InstallProps ) {
...
+			}
+			addNotice(
+				props.subscription.product_key,
+				errorMessage,
+				NoticeStatus.Error,
+				{
+					actions: [
+						{
+							label: __( 'Try again', 'woocommerce' ),
+							onClick: install,
```
Comment (KokkieH) [inline plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/actions/install.tsx]: I wonder if this is the best action...What if they click try again, just for it t fail again and give the same error?

An alternative might be to suggest they download and install manually instead, linking to the /downloads page on WCCOM?

File: plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/actions/install.tsx [hunk 2]
```diff
@@ -51,6 +51,44 @@ export default function Install( props: InstallProps ) {
 		);
 	};
 
+	const handleInstallError = ( error: any ) => {
```
Comment (KokkieH) [inline plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/actions/install.tsx:55]: My linter doesn't like this `any` :)

Should we perhaps add a eslint-ignore comment for this?

File: plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/actions/install.tsx (hunk 1 above)
Comment (KokkieH) [inline plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/actions/install.tsx]: I'm getting a linter error, `'install' was used before it was defined.`, so perhaps just switch the order of the functions?

**Review by KokkieH — APPROVED** [review]
Works as described.  I also ran `localStorage.setItem( 'debug', 'wc-admin:*' );` in the console so tracks events get logged there, and confirmed the event is being recorded.

Left some non-blocking comments in the code

File: plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/actions/install.tsx (hunk 1 above)
Comment (bor0) [reply plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/actions/install.tsx]: This is actually a really good suggestion!

WDYT of 3ad1f3c760284678ab91e88caaf8a149274f3516?

Comment (bor0) [reply plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/actions/install.tsx]: f6000649002433a156d014aba76ba5e606d5522c

File: plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/actions/install.tsx (hunk 2 above)
Comment (bor0) [reply plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/actions/install.tsx:55]: 2578e184b05d0e629e99833c4f5ddc9a22b8c586

File: plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/actions/install.tsx (hunk 1 above)
Comment (KokkieH) [reply plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/actions/install.tsx]: Perfect!

**Pull request merged**